runtime/commonTest/src/kotlinx/serialization/json/polymorphic/JsonPolymorphismExceptionTest.kt

```diff
@@ -0,0 +1,36 @@
+/*
+ * Copyright 2017-2020 JetBrains s.r.o. Use of this source code is governed by the Apache 2.0 license.
+ */
+
+package kotlinx.serialization.json.polymorphic
+
+import kotlinx.serialization.*
+import kotlinx.serialization.json.*
+import kotlinx.serialization.modules.*
+import kotlin.test.*
+
+class JsonPolymorphismExceptionTest : JsonTestBase() {
+
+    @Serializable
+    abstract class Base
+
+    @Serializable
+    @SerialName("derived")
+    class Derived(val nested: Nested = Nested()) : Base()
+
+    @Serializable
+    class Nested
+
+    @Test
+    fun testDecodingException() = parametrizedTest { useStreaming ->
+        val serialModule = SerializersModule {
+            polymorphic(Base::class) {
+                subclass<Derived>()
+            }
+        }
+
+        assertFailsWith<JsonDecodingException> {
```

Comment:
Don't you want to check the exception message?

Reply:
I usually avoid it